Fix an off by one error in Object-elf.C.  The code tried to access the
entry just past the last valid member of the regions_ vector.  This
led to a segfault because it picked up an invalid region pointer and
then tried to get the region name using it.


diff -Naurb symtabAPI.orig/src/dyninst/symtabAPI/src/Object-elf.C symtabAPI/src/dyninst/symtabAPI/src/Object-elf.C
--- symtabAPI.orig/src/dyninst/symtabAPI/src/Object-elf.C	2009-12-08 12:03:21.000000000 -0600
+++ symtabAPI/src/dyninst/symtabAPI/src/Object-elf.C	2012-10-29 13:36:12.000000000 -0500
@@ -1890,7 +1890,7 @@
             continue;
 
          Region *sec;
-         if(secNumber >= 1 && secNumber <= regions_.size()) {
+         if(secNumber >= 1 && secNumber < regions_.size()) {
             sec = regions_[secNumber];
          } else {
             sec = NULL;
